### PR TITLE
Start adding exception syntax support.

### PR DESCRIPTION
**Pull request by sofiafaro-da**
Counterpart to https://github.com/digital-asset/ghc/pull/59

This PR updates the ghc patch (adding support for the `exception` syntax) and adds a couple tests.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/tests/daml-test-files/ExceptionMultipleMessage.daml
Comment (cocreature) [inline compiler/damlc/tests/daml-test-files/ExceptionMultipleMessage.daml:9]: I think it might be good to test on the location at least once.

File: stack-snapshot.yaml
Comment (cocreature) [inline stack-snapshot.yaml:6]: let’s bump this to a the commit after merging the GHC pr.

File: compiler/damlc/tests/daml-test-files/ExceptionDataType.daml
Comment (cocreature) [inline compiler/damlc/tests/daml-test-files/ExceptionDataType.daml:65]: does regular braces record syntax work as well? if so, maybe worth adding test.

**Review by cocreature — APPROVED** [review]
Thanks!

File: stack-snapshot.yaml
Comment (sofiafaro-da) [reply stack-snapshot.yaml:6]: Yep! Waiting for the GHC PR to be merged before doing another round of ghc-lib updates (and including windows).

File: compiler/damlc/tests/daml-test-files/ExceptionDataType.daml
Comment (sofiafaro-da) [reply compiler/damlc/tests/daml-test-files/ExceptionDataType.daml:65]: Doing `exception E with { ... }` works, and I added a test for this.

Plain brackets `exception E { ... }` doesn't work, but it also doesn't work for templates AFAICT.

Comment (cocreature) [reply compiler/damlc/tests/daml-test-files/ExceptionDataType.daml:65]: Makes sense :+1:

**Pull request merged**
